### PR TITLE
Connect My Computer: Add ACL

### PR DESCRIPTION
**Pull request by gzdunek**
Connect My Computer feature should be visible only if the user has sufficient permissions (in this case it is an ability to create join tokens). To get that information, we need ACL.  
In the future, we can also use ACL to hide certain Servers/DBs/Kubes sections and limit the number of requests for resources in the search bar if the user can't list them. (gravitational/teleport.e#855)

I decided to share the logic between Web and Connect, so I extracted all code related to ACL from `lib/web/ui/usercontext.go` and moved it to `lib/services/useracl.go`. Three things I'm not sure about:
* Is `services` a good place for the shared code?
* It seemed to me that extracting the whole `NewUserACL` function is better than extracting only `hasAccess` and `newAccess` - we can use a more abstract function. OTOH we need to supply the parameter `desktopRecordingEnabled` that doesn't make much sense in Connect's context.   
*  Is it okay to have the json tags in the shared code? Theoretically they don't hurt us, but idk.

### CODE REVIEW

File: lib/services/useracl.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+		List:   hasAccess(roleSet, ctx, kind, types.VerbList),
+		Read:   hasAccess(roleSet, ctx, kind, types.VerbRead),
+		Edit:   hasAccess(roleSet, ctx, kind, types.VerbUpdate),
+		Create: hasAccess(roleSet, ctx, kind, types.VerbCreate),
+		Delete: hasAccess(roleSet, ctx, kind, types.VerbDelete),
+		Use:    hasAccess(roleSet, ctx, kind, types.VerbUse),
+	}
+}
+
+func NewUserACL(user types.User, userRoles RoleSet, features proto.Features, desktopRecordingEnabled bool) UserACL {
```
Comment (ravicious) [inline lib/services/useracl.go:115]: > It seemed to me that extracting the whole `NewUserACL` function is better than extracting only `hasAccess` and `newAccess` - we can use a more abstract function.

Yeah, I think it makes sense, given that `hasAccess` and `newAccess` have only been called from within the old equivalent of `NewUserACL` and we want to share the logic in `NewUserACL` between lib/web and lib/teleterm.

**Review by ravicious — APPROVED** [review]
LGTM.

I don't know the answers to the first and the last question. If I was you I'd ask directly on #dev-teleport – describe what I'm trying to achieve and what are my problems.

**Review by zmb3 — APPROVED** [review]

**Comment by gzdunek** [comment]
@ravicious I did a small change to `cluster.proto`. We should have `recorded_sessions` and `active_sessions` (as in Web) instead of `sessions`, since it is not clear which one we are referring to.

File: lib/teleterm/clusters/cluster.go [hunk 2]
```diff
@@ -113,6 +117,34 @@ func (c *Cluster) GetWithDetails(ctx context.Context) (*ClusterWithDetails, erro
...
+
+		roles, err := authClient.GetCurrentUserRoles(ctx)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		roleSet := services.NewRoleSet(roles...)
+		userACL := services.NewUserACL(user, roleSet, *pingResponse.ServerFeatures, false)
+
+		ACL = &api.ACL{
```
Comment (ravicious) [inline lib/teleterm/clusters/cluster.go:133]:
```suggestion
		acl = &api.ACL{
```

**Review by ravicious — APPROVED** [review]

File: lib/teleterm/clusters/cluster.go (hunk 2 above)
Comment (gzdunek) [reply lib/teleterm/clusters/cluster.go:133]: Done

**Pull request merged**